src/back-end/lib/db/opportunity/sprint-with-us.ts

```diff
@@ -1652,3 +1664,56 @@ export const addSWUOpportunityNote = tryDb<
   }
   return valid(dbResult.value);
 });
+
+export const submitIndividualQuestionEvaluations = tryDb<
+  [Id, SubmitQuestionEvaluationsWithNoteParams, AuthenticatedSession],
+  SWUOpportunity
+>(async (connection, id, evaluationParams, session) => {
+  const now = new Date();
+  await connection.transaction(async (trx) => {
+    for (const evaluationId of evaluationParams.evaluations) {
+      const [statusRecord] =
+        await connection<SWUTeamQuestionResponseEvaluationStatusRecord>(
+          "swuTeamQuestionResponseEvaluationStatuses"
+        )
+          .transacting(trx)
+          .insert(
+            {
+              id: generateUuid(),
+              teamQuestionResponseEvaluation: evaluationId,
+              createdAt: now,
+              createdBy: session.user.id,
+              status: SWUTeamQuestionResponseEvaluationStatus.Submitted,
+              note: evaluationParams.note
+            },
+            "*"
+          );
+
+      // Update evaluation root record
+      await connection<RawSWUTeamQuestionResponseEvaluation>(
+        "swuTeamQuestionResponseEvaluations"
+      )
+        .transacting(trx)
+        .where({ id: evaluationId })
+        .update(
+          {
+            updatedAt: now
+          },
+          "*"
+        );
+
+      if (!statusRecord) {
+        throw new Error("unable to update team question evaluation");
+      }
+    }
+  });
+
+  // TODO: Check if all individual proposals for a proposal have been submitted;
```

Comment:
Would this be a reasonable place to do this check? Feels a little messy because the functions in this file are specific to opportunities and we're trying to update proposal statuses.

Reply:
Is the check that all team question answers for the proposal have been submitted? I think that's okay. You could put it in the db resource for proposals and invoke it from here to keep it a bit cleaner and avoid this function from getting too long.